exercise_060_cluster_base/src/main/scala/akka/cluster/pi/ClusterStatusTracker.scala

```diff
@@ -0,0 +1,128 @@
+                           settings: Settings,
+                           timers: TimerScheduler[ClusterStatusTracker.Event]
+                           ) {
+  import ClusterStatusTracker._
+
+  private val cluster = Cluster(context.system)
+
+  private val thisHost = settings.config.getString("akka.remote.artery.canonical.hostname")
+
+  private val ledStrip = context.spawn(LedStripController(settings), "led-strip-controller")
+
+  timers.startTimerAtFixedRate(Tick, 1.seconds)
+
+  private val memberEventAdapter: ActorRef[MemberEvent] = context.messageAdapter(MemberChange)
+  Cluster(context.system).subscriptions ! Subscribe(memberEventAdapter, classOf[MemberEvent])
+
+  private val reachabilityAdapter: ActorRef[ReachabilityEvent] = context.messageAdapter(ReachabilityChange)
+  Cluster(context.system).subscriptions ! Subscribe(reachabilityAdapter, classOf[ReachabilityEvent])
+
+  private val roleLeadChangedAdapter: ActorRef[LeaderChanged] = context.messageAdapter(LeaderChange)
+  Cluster(context.system).subscriptions ! Subscribe(roleLeadChangedAdapter, classOf[LeaderChanged])
+
+  def running: Behavior[ClusterStatusTracker.Event] = Behaviors.receiveMessage { message =>
+    message match {
+      case ReachabilityChange(reachabilityEvent) =>
+        reachabilityEvent match {
+          case UnreachableMember(member) =>
+            ledStrip ! LedStripController.Unreachable(mapHostToLedId(member))
+          case ReachableMember(member) if member.status == Up =>
+            ledStrip ! LedStripController.Up(mapHostToLedId(member))
+          case ReachableMember(member) if member.status == WeaklyUp =>
+            ledStrip ! LedStripController.WeaklyUp(mapHostToLedId(member))
+        }
+      case MemberChange(memberChange) =>
+        memberChange match {
+          case MemberJoined(member) =>
+            ledStrip ! LedStripController.Joining(mapHostToLedId(member))
+          case MemberUp(member) =>
+            ledStrip ! LedStripController.Up(mapHostToLedId(member))
+          case MemberLeft(member) =>
+            ledStrip ! LedStripController.Leaving(mapHostToLedId(member))
+          case MemberExited(member) =>
+            ledStrip ! LedStripController.Exiting(mapHostToLedId(member))
+          case MemberRemoved(member, previousStatus) =>
+            ledStrip ! LedStripController.Removed(mapHostToLedId(member))
+          case MemberDowned(member) =>
+            ledStrip ! LedStripController.Down(mapHostToLedId(member))
+          case MemberWeaklyUp(member) =>
+            ledStrip ! LedStripController.WeaklyUp(mapHostToLedId(member))
+        }
+      case LeaderChange(LeaderChanged(None)) =>
+        ledStrip ! LedStripController.IsNoLeader
+      case LeaderChange(LeaderChanged(Some(leader))) if leader.host.getOrElse("") == thisHost =>
+        ledStrip ! LedStripController.IsLeader
+      case LeaderChange(LeaderChanged(Some(leader))) =>
+        ledStrip ! LedStripController.IsNoLeader
+      case Tick =>
```

Comment:
What do we use the `Tick` for?

Reply:
That's an artefact from some test I ran before. Will remove it